### PR TITLE
Use piece as property source in auto-move-report where possible

### PR DESCRIPTION
**Pull request by Cattlesquat**
Benkyo was trying to do a property computation in map auto-move-reporting that seemed reasonable, except strangely the movement reports use the Map rather than the GamePiece as the property source, so he couldn't use the piece properties.

This makes use of the piece as the property source. Since it has already been moved to the target (destination) map by the time the movement report is called, those properties remain in scope.

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java [hunk 1]
```diff
@@ -313,6 +315,10 @@ public Point getOldPosition() {
       return oldPosition;
     }
 
+    public List<GamePiece> getPieces() {
+      return new ArrayList<>(pieces);
+    }
+
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:321]: That kills me that you have to copy the whole list just to get the first element.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:321]: Even more hilarious is that, as far as I can tell, we NEVER ACTUALLY PUT MORE THAN ONE PIECE IN THIS!!!

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:324]: Which puts me into this newer/better idea -- I changed it to something that just returns the first piece in the array. That will at least save all the making-a-copy. Sometimes I think too much in terms of "future extensibility"

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:324]: So anyway that's checked in now.

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:324]: Why the first piece? Why not the last piece or the second piece? This is the reason that Map level auto-reports never used GamePiece properties, they are meaningless if you are dragging more than one piece.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:324]: Well that's what I *thought* except then Benkyo pointed out that it never actually does that. It generates one "MoveSummary" for EACH piece moved.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:324]: Or, well, no.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java:324]: So yes ... if we _want_ to make this work and "non-dodgy", then I'd probably put a checkbox on there telling the Map to report separately for every unit (and not group a single stack together in one autoreport line). And so then if you checked that box we could use the piece as the propertysource for each report. 

We can discuss whether it makes sense to go ahead and do that, or if we should leave folks who want "complex move reporting" to create their own logic in the pieces themselves (which is completely possible - just the "on ramp" from these things is a little hard to find)

**Comment by Cattlesquat** [comment]
User didn't end up pushing hard for us to do this, so I'm going to close this.

**Pull request closed without merging**